contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,283 @@
+      withdrawManager.addInput(priority, age, participant);
+      withdrawManager.addInput(priority, age2, msg.sender);
+    } else {
+      withdrawManager.addExitToQueue(msg.sender, childToken, rootToken, exitAmount, txHash, burnt, age);
+      withdrawManager.addInput(age, age, participant);
+    }
+  }
+
+  function verifyDeprecation(bytes calldata exit, bytes calldata inputUtxo, bytes calldata challengeData)
+    external
+    returns (bool)
+  {
+    PlasmaExit memory _exit = decodeExit(exit);
+    (uint256 age, address signer) = encodeInputUtxo(inputUtxo);
+    RLPReader.RLPItem[] memory referenceTxData = challengeData.toRlpItem().toList();
+
+    (, address childToken, address participant, bytes32 txHash,) = processExitTx(referenceTxData[10].toBytes());
+    require(
+      participant == signer,
+      "Challenge tx not signed by the party who signed the input UTXO to the exit"
+    );
+    require(
+      _exit.token == childToken,
+      "Challenge tx token doesnt match with exit token"
+    );
+    require(
+      _exit.txHash != txHash,
+      "Cannot challenge with the exit tx"
+    );
+    uint256 ageOfChallengeTx = withdrawManager.verifyInclusion(challengeData, 0, true /* verifyTxInclusion */);
+    (,,ageOfChallengeTx) = processReferenceTx(
+      referenceTxData[6].toBytes(), // receipt
+      referenceTxData[9].toUint(), // logIndex
+      signer,
+      childToken,
+      ageOfChallengeTx,
+      true /* isChallenge */);
+    return ageOfChallengeTx > age;
+  }
+
+  /**
+   * @notice Process the reference tx to start a MoreVP style exit
+   * @param receipt Receipt of the reference transaction
+   * @param logIndex Log Index to read from the receipt
+   * @param participant Either of exitor or a counterparty depending on the type of exit
+   */
+  function processReferenceTx(
+    bytes memory receipt,
+    uint256 logIndex,
+    address participant,
+    address childToken,
+    uint256 _age,
+    bool isChallenge)
+    internal
+    pure
+    returns(address rootToken, uint256 closingBalance, uint256 age)
+  {
+    RLPReader.RLPItem[] memory inputItems = receipt.toRlpItem().toList();
+    require(logIndex < MAX_LOGS, "Supporting a max of 10 logs");
+    inputItems = inputItems[3].toList()[logIndex].toList(); // select log based on given logIndex
+    require(
+      childToken == RLPReader.toAddress(inputItems[0]), // "address" (contract address that emitted the log) field in the receipt
+      "Reference and exit tx do not correspond to the same token"
+    );
+    bytes memory logData = inputItems[2].toBytes();
+    inputItems = inputItems[1].toList(); // topics
+    // now, inputItems[i] refers to i-th (0-based) topic in the topics array
+    // inputItems[0] is the event signature
+    rootToken = address(RLPReader.toUint(inputItems[1]));
+    // rootToken = RLPReader.toAddress(inputItems[1]); // investigate why this reverts
```

Comment:
I know why. data is 32 bytes and [this](https://github.com/hamdiallam/Solidity-RLP/blob/master/contracts/RLPReader.sol#L109) check fails.

Reply:
Then should I keep it this way or slice the data before calling `RLPReader.toAddress`?

Reply:
this one is fine.